framework/src/modules/sequence/sequence_module.ts

```diff
@@ -26,7 +36,47 @@ export class SequenceModule extends BaseModule {
 			},
 		},
 		default: {
-			balance: BigInt(0),
+			nonce: BigInt(0),
 		},
 	};
+
+	// eslint-disable-next-line class-methods-use-this
+	public async beforeTransactionApply({ tx, stateStore }: TransactionApplyInput): Promise<void> {
+		const senderAddress = getAddressFromPublicKey(tx.senderPublicKey);
+		const senderAccount = await stateStore.account.getOrDefault<Account<SequenceAccount>>(
+			senderAddress,
+		);
+
+		// Throw error when tx nonce is lower than the account nonce
+		if (tx.nonce < senderAccount.sequence.nonce) {
+			throw new NonceOutOfBoundsError(
```

Comment:
This error should be different in order to check in transaction pool.
ie) in transaction pool, the lower nonce is rejected immediately, but higher nonce is accepted.